worktree_status.go

```diff
@@ -370,7 +370,13 @@ func (w *Worktree) doAdd(path string, ignorePattern []gitignore.Pattern, skipSta
 		}
 	}
 
+	// Relative paths only; see: https://github.com/go-git/go-git/issues/1460
+	// filepath.IsAbs is hilariously bugged on Windows, check by hand
 	path = filepath.Clean(path)
+	if string(path[0]) == string(filepath.Separator) {
+		root := w.Filesystem.Root()
+		path, _ = filepath.Rel(root, filepath.Join(root, ".", path))
+	}
```

Comment:
@0xDEC0DE thanks for looking into this. This logic seems better suited in [go-billy](https://github.com/go-git/go-billy/), potentially as an `util` that receives a `Filesystem` and a path, returning a canonical path if the file exists.

We'd need to test this (in `go-billy`) for all built-in FS types.

Reply:
Not entirely sure I understand, as the use case here is creating a new file; but there's already `util.getUnderlyingAndPath`, which I suppose could use an exported version, would that be sufficient for this?